hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -512,7 +531,10 @@ private boolean isValidateNamespace(Namespace namespace) {
   public TableOperations newTableOps(TableIdentifier tableIdentifier) {
     String dbName = tableIdentifier.namespace().level(0);
     String tableName = tableIdentifier.name();
-    return new HiveTableOperations(conf, clients, fileIO, name, dbName, tableName);
+    HiveTableOperations hiveTableOperations =
+        new HiveTableOperations(conf, clients, fileIO, name, dbName, tableName);
+    fileIOCloser.put(hiveTableOperations, hiveTableOperations.io());
```

Comment:
Something I've wondered for a bit, is it possible to uplevel any of this logic to `BaseMetastoreCatalog` so that we avoid having every catalog which extends from that to have it's own Cache<TableOperations, FileIO>? 

Or does that not generalize? Not a blocker, but just wanted to make sure we at least thought through it to avoid duplication.

Reply:
I tried to move it to uplevel but I found some subclasses without a FileIO instance (`ViewAwareTableBuilder` and `HadoopCatalogTableBuilder`), if you have any suggestions to generalize it, I'm open to it.

Reply:
I have a TODO on my list to generalize this, so I'll take a look at this (hopefully this week)

Reply:
@amogh-jahagirdar I've opened https://github.com/apache/iceberg/pull/10893 to addresse this